bin/emsdk-inc.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/sh
+
+EMCC_DEBUG=${EMCC_DEBUG:-0}
+
+[ -z ${EMSDK} ] \
+  && {
+    echo "Missing EMSDK Environment variable.."
+    echo "Did you remember to run '/path/to/emsdk/emsdk_env.sh' ?"
+    [ "$(basename $0)" = "bash" ] \
+    || { 
+      echo "Terminating..."
+      exit 1
+    }
+  }
+
+#unset EMSCRIPTEN
+[ -z ${EMSCRIPTEN} ] \
+  && {
+    [ -d "${EMSDK}/upstream/emscripten" ] \
+    && {
+      export EMSCRIPTEN="${EMSDK}/upstream/emscripten"
+      echo "export EMSCRIPTEN = ${EMSCRIPTEN}"
+    } \
+    || {
+      echo "Missing directory: \${EMSDK}/upstream/emscripten"
+    }
+  } \
+  || {
+    echo "EMSCRIPTEN Environment OK"
+  }
+
+[ ${EMCC_DEBUG} -ge 1 ] \
+  && {
+    echo "Running make clean in build directory.."
+    make -C ./build clean
+  }
+
+get_boost_github() {
+  local SDK_PATH=$1
+  [ -z ${SDK_PATH} ] && { echo "get_boost_github: Missing SDK_PATH parameter..."; return 1; }
+
+  [ "$(basename ${SDK_PATH})" = "boost-sdk" ] \
+  && {
+    rm -fr ${SDK_PATH}
+    SDK_PATH=$(dirname ${SDK_PATH})
+  }
+  
+  [ -d ${SDK_PATH} ] || { echo "get_boost_github: Missing directory: ${SDK_PATH}"; return 1; }
+
+  echo "Downloading boost from GitHub..."
+  git clone --recursive https://github.com/boostorg/boost.git --branch "boost-1.72.0" "${SDK_PATH}/boost-sdk" \
+  || {
+    echo "Download failed.."
+    return 1
+  }
+  return 0
+}
+
+get_openssl_github() {
+  local SDK_PATH=$1
+  [ -z ${SDK_PATH} ] && { echo "get_openssl_github: Missing SDK_PATH parameter..."; return 1; }
+
+  [ "$(basename ${SDK_PATH})" = "openssl-sdk" ] \
+  && {
+    rm -fr ${SDK_PATH}
+    SDK_PATH=$(dirname ${SDK_PATH})
+  }
+  
+  [ -d ${SDK_PATH} ] || { echo "get_openssl_github: Missing directory: ${SDK_PATH}"; return 1; }
```

Comment:
Skip downloading from github or source and print message if destination folder already exists?  A clean install can be triggered by deleting the destination folder.

Reply:
I'm checking if a source archive is in `${SDK_PATH}`, and skips download is so..
That doesn't work if `github` is chosen.

I'm gonna chew on the `clean build/install` if `${SDK_PATH}` is deleted.

Reply:
I was thinking it could simply check for the existence of the folder, and skip the download if it exists.  That allows a fresh install by deleting the target folder, but skips downloading it if not necessary.  Thoughts?

Reply:
The `[ -d ${SDK_PATH} ]` is cheching for `external/monero-cpp-library/external`

So:
```
  [ -f ${SDK_PATH}/OpenSSL_1_1_1d.tar.gz ] \
  && {
    echo "openssl source is already in ${SDK_PATH}"
  } \
  || {
    echo "Downloading openssl source..."
    local DL_URL="https://github.com/openssl/openssl/archive/OpenSSL_1_1_1d.tar.gz"

    download_source ${DL_URL} ${SDK_PATH}  || return 1
  }
```

Reply:
I'm not sure doing stuff like this, without explicit asking for it, is the best idea..
But then again. Your the boss...

Reply:
There may be a misunderstanding.  The script should not delete from the file system, but would check if the destination folder exists.  If it exists, it would skip the download and print a message accordingly.  That way the script does not overwrite the destination folder or download unnecessarily.  Then if the library is freshly installed or if the user or another build script deletes the destination folder, this script would proceed to download.

Reply:
Okay, 
running `build-openssl-emscripten.sh` without a `source` or `github` parameter, 
checks if `openssl-sdk` exists. If not, check for source archive, download and extract as necessary, and continues it's merry way, building.